library/core/src/slice/mod.rs

```diff
@@ -143,6 +143,9 @@ impl<T> [T] {
     #[rustc_no_implicit_autorefs]
     #[inline]
     #[must_use]
+    #[ferrocene::annotation(
+        "this function is guaranteed to be constant-evaluated as the size of arrays is always available at compilation"
+    )]
     pub const fn len(&self) -> usize {
         ptr::metadata(self)
```

Comment:
hm, are you sure we can't verify this by using black_box?

Reply:
no difference :/